TTKDTrainingApp/src/components/HomeScreen.js

```diff
@@ -1,77 +1,58 @@
-/*
-    Placeholder screen to allow for creation of components directory
-    Will be either repurposed or deleted as development starts
-*/
+import React from 'react';
+import {StyleSheet, View, FlatList} from 'react-native';
+import {useSelector} from 'react-redux';
+import {isEmpty} from 'lodash';
 
-import React, {useState} from 'react';
-import {StyleSheet, View, Text, TouchableOpacity} from 'react-native';
-import Icon from 'react-native-vector-icons/FontAwesome';
+import {getSubCategories, getCategoryContent} from './../redux/selectors';
 
-import Categories from './Categories';
-import {getCategory, getCategories} from '../utilities/categories.js';
+import CategoryCard from './CategoryCard';
+import ContentCard from './ContentCard';
 
-const HomeScreen = () => {
-  const [categoryId, setCategoryId] = useState(null);
-  const category = getCategory(categoryId);
-
-  const renderBackButton = (
-    <View style={styles.backButtonContainer}>
-      {category && (
-        <TouchableOpacity
-          style={styles.backButton}
-          onPress={() => setCategoryId(category.parent_id)}>
-          <Icon
-            name="angle-left"
-            size={20}
-            style={styles.backButtonIcon}
-            backgroundColor={null}
-            onPress={() => setCategoryId(category.parent_id)}
-          />
-          <Text style={styles.backButtonText}>{' Back'}</Text>
-        </TouchableOpacity>
-      )}
-    </View>
+const HomeScreen = ({navigation}) => {
+  const category = navigation.getParam('category');
+  const categoryId = category ? category.id : null;
+  const categoryContent = useSelector(state =>
+    getCategoryContent(state, categoryId),
   );
-
-  const renderTitle = (
-    <View style={styles.title}>
-      <Text style={styles.titleText}>
-        {categoryId ? category.category_name : 'TTKD Home'}
-      </Text>
-    </View>
+  const subCategories = useSelector(state =>
+    getSubCategories(state, categoryId),
```

Comment:
do we have use cases when there are subcategories that have subcategories? Or the use case is just main category and then only one level down of categories can exist ?

Reply:
Subcategories can exist here. When someone taps a `CategoryCard` that is a subCategory, we push the `HomeScreen` component onto the navigation stack and set the current category to the selected `categoryId` of the pressed `CategoryCard`

Reply:
I realized as I was typing ^^ that it may be a bit hard to follow, so here's an example:
category 1 is the main category, category 2 has 1 as its parent, and category 3 has 2 as its parent, so:
- `HomeScreen` -> categoryId is 1, shows category 2 as a subcategory
- Tap category 2, push `HomeScreen` on the stack with category 2
- `HomeScreen` categoryId is 2, shows category 3 as subcategory
- Tap category 3, push `HomeScreen` on the stack with category 3

Reply:
ah i see!